config/application.rb

```diff
@@ -44,8 +44,12 @@ class Application < Rails::Application
     config.action_controller.forgery_protection_origin_check = false
     # Make `form_with` generate non-remote forms. Defaults true in Rails 5.1 to 6.0
     config.action_view.form_with_generates_remote_forms = false
+
     # Make Ruby preserve the timezone of the receiver when calling `to_time`.
     config.active_support.to_time_preserves_timezone = false
+    # Use a modern approved hashing function
+    # config.active_support.hash_digest_class = OpenSSL::Digest::SHA256
+    ActiveSupport::Digest.hash_digest_class = OpenSSL::Digest::SHA256 # option above should work with Rails 6.x
```

Comment:
What are your thoughts around `ActiveSupport::Digest.hexdigest` (https://github.com/rails/rails/blob/5-2-stable/activesupport/lib/active_support/digest.rb).  This is the method:

```ruby
def hexdigest(arg)
  hash_digest_class.hexdigest(arg)[0...32]
end
```

`OpenSSL::Digest::MD5` generates a string of length 32, but `OpenSSL::Digest::SHA256` generates a string of length 64.  So truncating it has no impact on the effectiveness of MD5, but might for SHA256.  I'm not sure we have a choice in the matter since MD5 is a non-starter, but wanted to still drop a note here.  Maybe we can look at Rails 6.x and see what other changes went into ensuring that SHA256 worked.

Reply:
The change already went into 5.2 and I couldn't spot any complaints about hash length. I also went over same pieces of code in Rails 7 for the purpose you describe. One issue was the memcache store that we switched to and is mostly broken regardless of hash size. I have rails/rails#45245 for that but still it works well enough for our use cases.